### PR TITLE
acceptance: allow docker tests to run

### PR DESCRIPTION
**Pull request by madelynnblue**
Cache the builder image, which is now just one image with all langs.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4113)

### CODE REVIEW

File: build/circle-deps.sh [hunk 1]
```diff
@@ -112,19 +132,10 @@ ls -lah "${builder_dir}"
...
-  find "${builder_dir}" -not -path "${builder}" -type f -delete  
-  if [[ ! -e "${builder}" ]]; then
-    time docker pull "cockroachdb/builder:${tag}"
-    docker tag -f "cockroachdb/builder:${tag}" "cockroachdb/builder:latest"
-    time docker save "cockroachdb/builder:${tag}" > "${builder}"
-  else
-    time docker load -i "${builder}"
-    docker tag -f "cockroachdb/builder:${tag}" "cockroachdb/builder:latest"
-  fi
+fetch_docker "builder" ${tag}
```
Comment (tamird) [inline build/circle-deps.sh]: inline the tag here too?

File: build/circle-deps.sh [hunk 2]
```diff
@@ -112,19 +132,10 @@ ls -lah "${builder_dir}"
...
+
+if is_shard 0; then
+  fetch_docker "postgres-test" "20160203-140220"
```
Comment (tamird) [inline build/circle-deps.sh]: can you add a comment here that points to the dockerfile?

File: acceptance/c_test.go [hunk 3]
```diff
@@ -23,11 +23,10 @@ import (
 	"testing"
 )
 
-// TestC connects to a cluster with C.
-func TestC(t *testing.T) {
-	t.Skip("https://github.com/cockroachdb/cockroach/issues/3826")
-	testDockerSuccess(t, "c", []string{"/bin/sh", "-c", strings.Replace(c, "%v", "3", 1)})
-	testDockerFail(t, "c", []string{"/bin/sh", "-c", strings.Replace(c, "%v", "a", 1)})
+// TestDockerC connects to a cluster with C.
```
Comment (tamird) [inline acceptance/c_test.go]: I think "connects to a cluster with" is outdated on all these functions.

Comment (tamird) [inline acceptance/c_test.go]: But I also think removing these comments would be fine.

**Comment by tamird** [comment]
LGTM

**Comment by petermattis** [comment]
Review status: 0 of 7 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[build/circle-deps.sh, line 15 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4113#-K9cc13MQVipqAcISV2D:-K9cc13MQVipqAcISV2E:1358344271) ([raw file](https://github.com/cockroachdb/cockroach/blob/fc5019cbc952caba3834c48709df08312c72ac40/build/circle-deps.sh#L15)):*
For a slight bit of cleanliness: you could tag the above 3 variables as `local` (e.g. `local image=${1}`).

---

*[build/circle-deps.sh, line 20 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4113#-K9cchqMD0z5OVTSPtmI:-K9cchqMD0z5OVTSPtmJ:-1343352168) ([raw file](https://github.com/cockroachdb/cockroach/blob/fc5019cbc952caba3834c48709df08312c72ac40/build/circle-deps.sh#L20)):*
I think this removes any file not named `${builder}`, which isn't what we want. I think we have to add on a `-name "${image}:*"` filter to the `find` command.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4113)*

**Comment by tamird** [comment]
Are we just not doing this?

**Comment by mjibson** [comment]
The tests were failing with 5m timeouts for an unknown reason. I need to take another look.

**Comment by mjibson** [comment]
(Also I misclicked and accidentally closed this.)

**Pull request closed without merging**